### ISSUE TITLE
macOS use login-keychain to avoid sudo

### ISSUE THREAD

**Issue by andrewmackrodt** [mlebkowski/nassau-https-proxy#8]
Including `nassau-https-proxy` on a macOS system assumes the certificate should be added to the system keychain. Ideally this should default to `false` or allow an environment variable to disable this behaviour.

However, it installs itself to the system keychain requiring sudo access. It would be better to use the login-keychain, e.g.

```sh
keychain=$(security login-keychain | perl -pe 's/^.+?"|"$//g')
security add-trusted-cert -p ssl -k "$keychain" "$certificate"
```

**Comment by mlebkowski** [comment]
Did you confirm that this works from login keychain?

**Comment by andrewmackrodt** [comment]
It's the method I used to use in other projects but I've been primarily using Linux the last couple of years so it may have changed between High Sierra and newer revisions.

**Comment by mlebkowski** [comment]
I confirmed, it seems to work, but it asks for confirmation (GUI) either way on `10.14.x`. And it failed to generate root CA on my system (I use the tool in docker nowdays), so I’m not sure I want to make any changes, I have to sleep on it

**Comment by andrewmackrodt** [comment]
The interactive GUI is IMO a useful addition in this case. I've had macOS colleagues experience issues when including the package as a dependency as npm's postinstall is running in non-interactive mode, i.e. the sudo command fails.

**Comment by mlebkowski** [comment]
@andrewmackrodt, take a look at fa90f79a41961e3bbe816443bc77337814ea4e17

As for the issue at hand:

1) I don’t believe this step *has* to succeed. You can do it manually.
2) The package isn’t meant to be installed as a dependency and I doubt anyone would use it this way

If that works for you I can merge & publish

**Comment by andrewmackrodt** [comment]
Thanks for this! Will test on my mac and reply later. Regarding your dependency point, it can be quite useful as a dev dependency for cases where developing against a third-party service requires an https redirection domain, e.g. starting the proxy via a package.json script. This was a request from one of my front-end colleagues.

There are other projects such as `dev-proxy` but I found `nassau-https-proxy` to be superior in that it creates it's own CA and sensibly places certificates outside of the node_modules folder making it ideal for multi-project use. However, my macOS colleagues reported that `npm install` was failing due to the library requesting `sudo` access.

**Comment by mlebkowski** [comment]
fixed in fa90f79a41961e3bbe816443bc77337814ea4e17
